XDMA/linux-kernel/xdma/xdma_netdev.c

```diff
@@ -83,18 +89,53 @@ int xdma_netdev_close(struct net_device *ndev)
         return 0;
 }
 
+#if DEBUG_ONE_QUEUE_TSN_
+void dump_buffer(unsigned char* buffer, int len)
+{
+        int i = 0;
+        pr_err("[Buffer]");
+        pr_err("%02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x",
+        buffer[i+0] & 0xFF, buffer[i+1] & 0xFF, buffer[i+2] & 0xFF, buffer[i+3] & 0xFF,
+        buffer[i+4] & 0xFF, buffer[i+5] & 0xFF, buffer[i+6] & 0xFF, buffer[i+7] & 0xFF,
+        buffer[i+8] & 0xFF, buffer[i+9] & 0xFF, buffer[i+10] & 0xFF, buffer[i+11] & 0xFF,
+        buffer[i+11] & 0xFF, buffer[i+13] & 0xFF, buffer[i+14] & 0xFF, buffer[i+15] & 0xFF);
+
+        i = 16;
+        pr_err("%02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x",
+        buffer[i+0] & 0xFF, buffer[i+1] & 0xFF, buffer[i+2] & 0xFF, buffer[i+3] & 0xFF,
+        buffer[i+4] & 0xFF, buffer[i+5] & 0xFF, buffer[i+6] & 0xFF, buffer[i+7] & 0xFF,
+        buffer[i+8] & 0xFF, buffer[i+9] & 0xFF, buffer[i+10] & 0xFF, buffer[i+11] & 0xFF,
+        buffer[i+11] & 0xFF, buffer[i+13] & 0xFF, buffer[i+14] & 0xFF, buffer[i+15] & 0xFF);
+
+        i = 32;
+        pr_err("%02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x",
+        buffer[i+0] & 0xFF, buffer[i+1] & 0xFF, buffer[i+2] & 0xFF, buffer[i+3] & 0xFF,
+        buffer[i+4] & 0xFF, buffer[i+5] & 0xFF, buffer[i+6] & 0xFF, buffer[i+7] & 0xFF,
+        buffer[i+8] & 0xFF, buffer[i+9] & 0xFF, buffer[i+10] & 0xFF, buffer[i+11] & 0xFF,
+        buffer[i+11] & 0xFF, buffer[i+13] & 0xFF, buffer[i+14] & 0xFF, buffer[i+15] & 0xFF);
+
+        pr_err("\n");
+}
+#endif
+
 netdev_tx_t xdma_netdev_start_xmit(struct sk_buff *skb,
                 struct net_device *ndev)
 {
         struct xdma_private *priv = netdev_priv(ndev);
         struct xdma_dev *xdev = priv->xdev;
+        struct xdma_pci_dev *xpdev = dev_get_drvdata(&priv->pdev->dev);
         int padding = 0;
         u32 w;
+        u32 sys_count_low;
+        u16 frame_length;
         dma_addr_t dma_addr;
+        struct tx_metadata* tx_metadata;
 
         /* Check desc count */
         netif_stop_queue(ndev);
-        pr_err("xdma_netdev_start_xmit\n");
+#if DEBUG_ONE_QUEUE_TSN_
+        pr_err("xdma_netdev_start_xmit(skb->len : %d)\n", skb->len);
+#endif
         padding = (skb->len < ETH_ZLEN) ? (ETH_ZLEN - skb->len) : 0;
         skb->len += padding;
```

Comment:
이번 변경사항은 아니지만 여기서 padding 변수는 굳이 필요 없어보입니다
```suggestion
        skb->len = (skb->len < ETH_ZLEN) ? ETH_ZLEN : skb->len;
```

Reply:
수정했습니다.